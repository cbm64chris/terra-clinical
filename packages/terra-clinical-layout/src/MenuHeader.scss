/* stylelint-disable selector-class-pattern */
@import '~terra-mixins';

@include cerner-mock-theme {
  --layout-menu-header-background-color: plum;
  --layout-menu-header-border-bottom: 1px solid #EDEEEE;
  --layout-menu-header-border-top: 1px solid #BCBFC0;
  --layout-menu-header-color: blue;
  --layout-menu-header-font-size: 14px;
  --layout-menu-header-height: 40px;
}

:local {
  .header {
    align-items: center;
    background: var(--layout-menu-header-background-color, #E8E9EA);
    border-bottom: var(--layout-menu-header-border-bottom, 1px solid #EDEEEE);
    border-bottom: var(--layout-menu-header-border-top, 1px solid #BCBFC0);
    color: var(--layout-menu-header-color, #4E5558);
    display: flex;
    font-size: var(--layout-menu-header-font-size, 12px);
    height: var(--layout-menu-header-height, 29px);
    justify-content: space-between;
    position: relative;
    width: 100%;
  }

  .header-content-end {
    display: flex;
    flex: 0 0 auto;
    justify-content: flex-end;
    max-height: 100%;
    min-width: 0;
    position: relative;
  }

  .header-content-body {
    display: flex;
    flex: 1 1 auto;
    justify-content: flex-start;
    max-height: 100%;
    min-width: 0;
    padding: 5px;
    padding-left: 10px;
    position: relative;
  }

  .icon {
    height: 14px;
    margin-top: 3px;
    width: 14px;
  }

  .unlock {
    transform: rotate(90deg);
    transform-origin: left top 50%;
  }

  .header-display {
    margin: 0;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    word-break: normal;
    word-wrap: normal;
  }

  .header-button {
    outline: none;
  }
}