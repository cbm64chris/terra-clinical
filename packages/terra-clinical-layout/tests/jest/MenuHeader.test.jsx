import React from 'react';

import MenuHeader from '../../src/_MenuHeader';

describe('MenuHeader', () => {
  it('should render without optional props', () => {
    const result = shallow((
      <MenuHeader />
    ));
    expect(result).toMatchSnapshot();
  });

  it('should render with provided props', () => {
    const result = shallow((
      <MenuHeader
        text="TEST TEXT"
        togglePin={() => {}}
        isPinned
      >
        <div>Child</div>
      </MenuHeader>
    ));

    expect(result).toMatchSnapshot();
  });

  it('should render when not pinned', () => {
    const result = shallow((
      <MenuHeader
        text="TEST TEXT"
        togglePin={() => {}}
      >
        <div>Child</div>
      </MenuHeader>
    ));

    expect(result).toMatchSnapshot();
  });
});
