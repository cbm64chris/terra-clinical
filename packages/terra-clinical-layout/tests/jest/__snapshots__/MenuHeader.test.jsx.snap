// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuHeader should render when not pinned 1`] = `
<div
  className="header"
>
  <div
    className="header-content-body"
  >
    <h3
      className="header-text"
    >
      TEST TEXT
    </h3>
  </div>
  <div
    className="header-content-end"
  >
    <Button
      className="header-button"
      icon={
        <IconPin
          className="icon unlock"
        />
      }
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isReversed={false}
      onClick={[Function]}
      type="button"
      variant="link"
    />
  </div>
</div>
`;

exports[`MenuHeader should render with provided props 1`] = `
<div
  className="header"
>
  <div
    className="header-content-body"
  >
    <h3
      className="header-text"
    >
      TEST TEXT
    </h3>
  </div>
  <div
    className="header-content-end"
  >
    <Button
      className="header-button"
      icon={
        <IconPin
          className="icon"
        />
      }
      isBlock={false}
      isCompact={false}
      isDisabled={false}
      isReversed={false}
      onClick={[Function]}
      type="button"
      variant="link"
    />
  </div>
</div>
`;

exports[`MenuHeader should render without optional props 1`] = `
<div
  className="header"
>
  <div
    className="header-content-body"
  >
    <h3
      className="header-text"
    />
  </div>
  <div
    className="header-content-end"
  />
</div>
`;
